### PR TITLE
Return duplicade HTTP header fields in remote as array.

### LINKED ISSUE #234: Link-header gets overwritten by two values
This URL has two Link-headers, and I am interested in the first: https://www.svenknebel.de/posts/2017/5/13/

```
HTTP/1.1 401 UNAUTHORIZED
Server: nginx/1.9.10
Date: Sun, 21 May 2017 15:49:14 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 13
Connection: close
Link: <https://www.svenknebel.de/post_token>; rel="token_endpoint"
Link: </webmention>; rel="webmention"
WWW-Authenticate: Bearer

auth required
```

The `remote::head()` overwrites the Link header when it sees a second one.

```
  ["headers"]=>
  array(7) {
    ["Server"]=>
    string(12) "nginx/1.9.10"
    ["Date"]=>
    string(29) "Sun, 21 May 2017 16:17:00 GMT"
    ["Content-Type"]=>
    string(24) "text/html; charset=utf-8"
    ["Content-Length"]=>
    string(2) "13"
    ["Connection"]=>
    string(10) "keep-alive"
    ["Link"]=>
    string(31) "</webmention>; rel="webmention""
    ["WWW-Authenticate"]=>
    string(6) "Bearer"
  }
```

### PR DESCRIPTION
**Pull request by Zegnat**
Fixes #234 per discussion in #235. No header field name normalisation,
but no more loss of data.

### CODE REVIEW

File: lib/remote.php [hunk 1]
```diff
@@ -155,7 +155,15 @@ protected function header($curl, $header) {
 
     if(!empty($parts[0]) && !empty($parts[1])) {
       $key = array_shift($parts);
-      $this->headers[$key] = implode(':', $parts);
+
+      if (empty($this->headers[$key])) {
```
Comment (lukasbestle) [inline lib/remote.php]: Please use `!isset()` here. `empty()` behaves quite strangely.

**Review by lukasbestle — CHANGES_REQUESTED** [review]
Looks good generally, thanks!
We won't merge this in 2.4.2 though as this is a breaking change.

File: lib/remote.php (hunk 1 above)
Comment (Zegnat) [inline lib/remote.php]: Done, that was silly of me, I know it would have failed on strings like `0`.

**Review by lukasbestle — APPROVED** [review]
Thanks, looks good now.

**Pull request closed without merging**